sentinel.go

```diff
@@ -335,6 +370,41 @@ func (c *sentinelClient) listWatch(cc conn) (master string, sentinels []string,
 			sentinels = append(sentinels, fmt.Sprintf("%s:%s", m["ip"], m["port"]))
 		}
 	}
+
+	// we return random slave address instead of master
+	if c.readonly {
+		replicaResponses, err := resp[1].ToArray()
+		if err != nil {
+			return "", nil, err
+		}
+
+		eligibleReplicas := make([]RedisMessage, 0, len(replicaResponses))
+		// eliminate replicas with s_down condition
+		for i := range replicaResponses {
+			replicaResponseMap, err := replicaResponses[i].AsMap()
+			if err != nil {
+				continue
+			}
+			if _, containsSDownTime := replicaResponseMap["s-down-time"]; !containsSDownTime {
+				eligibleReplicas = append(eligibleReplicas, replicaResponses[i])
+			}
+		}
+
+		if len(eligibleReplicas) == 0 {
+			return "", nil, fmt.Errorf("not enough ready replicas")
+		}
+
+		// choose a replica randomly
+		randomReplicaIndex := rand.Intn(len(eligibleReplicas))
+		m, err := eligibleReplicas[randomReplicaIndex].AsStrMap()
+		if err != nil {
```

Comment:
We can skip the error checking here, since we have checked them above (if we also use AsStrMap above).